hybrid agent release (#16238)
newrelic/docs-website: src/content/docs/release-notes/mobile-release-notes/capacitor-release-notes/capacitor-agent-134.mdx

---
subject: Capacitor agent
releaseDate: '2024-02-15'
version: 1.3.4
downloadLink: 'https://www.npmjs.com/package/@newrelic/newrelic-capacitor-plugin/v/1.3.4'
---

> # New in this release
> We have made some updates to our system to improve its functionality. These changes include:

Reviewer: ```suggestion
# New in this release
In this release, we've made the following changes: 
```

> - Fixed a bug in the fetch instrumentation where customer options were inadvertently removed when no headers were specified. Now, options will be preserved even when headers are absent.
> - Resolved a bug in the XMLHttpRequest instrumentation that was throwing errors when headers were added by the instrumentation. This issue has been fixed to prevent any unexpected errors in such cases.
> - Addressed an issue that resulted in app crashes when an invalid URL was encountered in the capacitor plugin. To mitigate this, a valid URL checker has been implemented to ensure that mobilerequest events are created only with valid URLs.
> 
> These updates are intended to improve overall performance and functionality of our system. We appreciate your patience and understanding as we work to enhance your experience.

Reviewer: ```suggestion
- Fixed a bug in the fetch instrumentation where customer options were inadvertently removed when no headers were specified. Now, options will be preserved even when headers are absent.
- Resolved a bug in the `XMLHttpRequest instrumentation` that was throwing errors when headers were added by the instrumentation. This issue has been fixed to prevent any unexpected errors in such cases.
- Addressed an issue that resulted in app crashes when an invalid URL was encountered in the capacitor plugin. To mitigate this, a valid URL checker has been implemented to ensure that `mobilerequest` events are created only with valid URLs.

These updates are intended to improve the overall performance and functionality of our agent. We appreciate your patience and understanding as we work to enhance your experience.
```
